pkg/appgw/frontend_listeners.go

```diff
@@ -88,21 +90,23 @@ func (c *appGwConfigBuilder) getPublicIPID() *string {
 		} else {
 			jsonConfigs = append(jsonConfigs, string(jsonConf))
 		}
-		// Either PublicIPAddress is nil or PrivateIPAddress; never both present never both nil;
-		if ip.ApplicationGatewayFrontendIPConfigurationPropertiesFormat != nil && ip.PublicIPAddress != nil {
-			publicIPID = ip.ID
+
+		if usePrivateIP, _ := strconv.ParseBool(envVariables.UsePrivateIP); usePrivateIP {
+			if ip.ApplicationGatewayFrontendIPConfigurationPropertiesFormat != nil && ip.PrivateIPAddress != nil {
```

Comment:
I wonder if we could use `if ip.ApplicationGatewayFrontendIPConfigurationPropertiesFormat != nil` outside of this current if block.